File: javaProject.IntelliJ/LAb1_VendMachine/src/Beverage.java
public class Beverage {

    private final String name;
    private final double price;
    private int stock;
    private int timesSelected;

    public Beverage(String name, double price, int stock){
        this.name = name;
        this.price = price;
        this.stock = stock;
        this.timesSelected = 0;
    }

    public void decreaseStock(){
        if(stock > 0)
            --this.stock;
    }

    public void increaseTimesSelected(){++this.timesSelected;}
    public void setStock(int stock){this.stock = stock;}

    public String getName()  {return name;}
    public double getPrice()  {return price;}
    public int getStock()       {return stock;}
    public int getTimesSelected() {return timesSelected;}
}

File: C++OOP/OOPTesting/Beverage.cpp
#include "Beverage.h"

Beverage::Beverage(string name, double price, int stock)
    : name(name), price(price), stock(stock) {}

void Beverage::setName(string name){this->name = name;}
void Beverage::setStock(int stock){this->stock = stock;}
void Beverage::setPrice(double price){this->price = price;}

void Beverage:: decreaceStock(){
    if(this->stock <= 0)
        return;
    this->stock--;
}

int Beverage::getStock(){return stock;}
string Beverage::getName(){return name;}
double Beverage::getPrice(){return price;}
